lib/alchemy/dragonfly/processors/thumbnail.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require "dragonfly/image_magick/commands"
+require "dragonfly/image_magick/processors/thumb"
+
+module Alchemy
+  module Dragonfly
+    module Processors
+      class Thumbnail
+        include ::Dragonfly::ParamValidators
+
+        def initialize
+          @dragonfly_thumb = ::Dragonfly::ImageMagick::Processors::Thumb.new
+        end
+
+        # resizing and crop for images
+        # due to a missing ImageMagick parameter animated GIFs were broken with the default
+        # Dragonfly Thumb processor
+        def call(content, geometry, opts = {})
+          validate!(opts["format"], &is_word)
+          validate!(opts["frame"], &is_number)
+
+          geometry_args = @dragonfly_thumb.args_for_geometry(geometry)
+
+          # resize all frames in a GIF
+          # @link https://imagemagick.org/script/command-line-options.php#coalesce
+          # @link https://imagemagick.org/script/command-line-options.php#deconstruct
+          if content.mime_type == "image/gif"
+            geometry_args = "-coalesce #{geometry_args} -deconstruct"
+          end
+
+          ::Dragonfly::ImageMagick::Commands.convert(
+            content,
+            geometry_args,
+            {
+              "format" => opts["format"],
+              "frame" => opts["frame"],
+            }
+          )
+        end
```

Comment:
Could you just inherit from `Dragonfly::ImageMagick::Processors::Thumb` and override the `args_for_geometry`   method, call `super` in it and add the gif related code? I think this should reduce the amount of internals we need to reimplement here.